airflow/api_fastapi/execution_api/datamodels/taskinstance.py

```diff
@@ -181,6 +182,7 @@ class DagRun(BaseModel):
     data_interval_end: UtcDateTime | None
     start_date: UtcDateTime
     end_date: UtcDateTime | None
+    clear_number: int
```

Comment:
it's odd to `clear_number` in the DagRun datamodel!

What do we need this for? Can we get it from somewhere else?

Reply:
It is a part of the current DagRun model :) https://github.com/apache/airflow/blob/02d83b0768ff8ad3024b3a42cf8829789867861f/airflow/models/dagrun.py#L160

We need it to properly generate DR uuid, so that events from different physical executions of a dag run aren't mixed up: https://github.com/apache/airflow/blob/90eae569db448bf24afd4b14505f055100c1193e/providers/src/airflow/providers/openlineage/plugins/adapter.py#L122

Reply:
I should have clarified better: I meant `clear_number` on the `DagRun` Runtime model feels odd i.e. it isn't required for it.

Since `logical_date` can now be `None` too based on the link below, I think, you might need to refactor the logic for generating DR uuid.

https://lists.apache.org/thread/cknldkl9pmmzr1q7ot67wborzznlwrtv

Reply:
Conceptually (in OpenLineage) if you clear and rerun a DAG run, the two runs (before and after the clear) are treated as entirely different objects. I beleive this is kind of what we want to do in Airflow in the long run (similar to how we added TaskInstanceHistory), but before that happens, OL needs clear_number to distinguish _logically_ different runs that have the same run ID.

Reply:
I agree that this needs to change - but I would leave that for followup PR after the actual change in Airflow behavior will get merged https://github.com/apache/airflow/pull/45732